homeassistant/components/influxdb/sensor.py

```diff
@@ -121,63 +142,30 @@ def validate_query_format_for_version(conf: Dict) -> Dict:
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+def setup_platform(hass, conf, add_entities, discovery_info=None):
     """Set up the InfluxDB component."""
-    use_v2_api = config[CONF_API_VERSION] == API_VERSION_2
-    queries = None
-
-    if use_v2_api:
-        influx_conf = {
-            "url": config[CONF_URL],
-            "token": config[CONF_TOKEN],
-            "org": config[CONF_ORG],
-        }
-        bucket = config[CONF_BUCKET]
-        queries = config[CONF_QUERIES_FLUX]
-
-        for v2_query in queries:
-            if CONF_BUCKET not in v2_query:
-                v2_query[CONF_BUCKET] = bucket
-
-    else:
-        influx_conf = {
-            "database": config[CONF_DB_NAME],
-            "verify_ssl": config[CONF_VERIFY_SSL],
-        }
-
-        if CONF_USERNAME in config:
-            influx_conf["username"] = config[CONF_USERNAME]
-
-        if CONF_PASSWORD in config:
-            influx_conf["password"] = config[CONF_PASSWORD]
-
-        if CONF_HOST in config:
-            influx_conf["host"] = config[CONF_HOST]
-
-        if CONF_PATH in config:
-            influx_conf["path"] = config[CONF_PATH]
-
-        if CONF_PORT in config:
-            influx_conf["port"] = config[CONF_PORT]
-
-        if CONF_SSL in config:
-            influx_conf["ssl"] = config[CONF_SSL]
-
-        queries = config[CONF_QUERIES]
+    try:
+        influx = get_influx_connection(conf, test_read=True)
+    except ConnectionError as exc:
+        _LOGGER.error(RETRY_MESSAGE, exc)
+        event_helper.call_later(
+            hass,
+            RETRY_INTERVAL,
+            lambda _: setup_platform(hass, conf, add_entities, discovery_info),
+        )
+        return True
```

Comment:
Nothing is checking this return value.

Reply:
Oh, I'll remove that then. Was using the same error-handling code, didn't realize that difference.